convert2rhel/actions/conversion/preserve_only_rhel_kernel.py

```diff
@@ -0,0 +1,207 @@
+                        description="Conflict of kernels: One of the installed kernels"
+                        " has the same version as the latest RHEL kernel.",
+                    )
+                    pkghandler.handle_no_newer_rhel_kernel_available()
+                    kernel_update_needed = True
+        if kernel_update_needed:
+            pkghandler.update_rhel_kernel()
+
+
+class VerifyRhelKernelInstalled(actions.Action):
+    id = "VERIFY_RHEL_KERNEL_INSTALLED"
+
+    def run(self):
+        """Verify that the RHEL kernel has been successfully installed and raise an ERROR if not"""
+        loggerinst.info("Verifying that RHEL kernel has been installed")
+        if not pkghandler.is_rhel_kernel_installed():
+            self.set_result(
+                level="ERROR",
+                id="NO_RHEL_KERNEL_INSTALLED",
+                title="No RHEL kernel installed",
+                description="There is no RHEL kernel installed on the system.",
+                remediations="Verify that the repository used for installing kernel contains RHEL packages.",
+            )
+        else:
+            loggerinst.info("RHEL kernel has been installed.")
+            self.add_message(
+                level="INFO",
+                id="RHEL_KERNEL_INSTALLED",
+                title="RHEL kernel installed",
+                description="The RHEL kernel has been installed successfully.",
+            )
+
+
+class FixInvalidGrub2Entries(actions.Action):
+    id = "FIX_INVALID_GRUB2_ENTRIES"
+
+    def run(self):
+        """
+        On systems derived from RHEL 8 and later, /etc/machine-id is being used to identify grub2 boot loader entries per
+        the Boot Loader Specification.
+        However, at the time of executing convert2rhel, the current machine-id can be different from the machine-id from the
+        time when the kernels were installed. If that happens:
+        - convert2rhel installs the RHEL kernel, but it's not set as default
+        - convert2rhel removes the original OS kernels, but for these the boot entries are not removed
+        The solution handled by this function is to remove the non-functioning boot entries upon the removal of the original
+        OS kernels, and set the RHEL kernel as default.
+        """
+        if system_info.version.major < 8 or system_info.arch == "s390x":
+            # Applicable only on systems derived from RHEL 8 and later, and systems using GRUB2 (s390x uses zipl)
+            return
+
+        loggerinst.info("Fixing GRUB boot loader entries.")
+
+        machine_id = utils.get_file_content("/etc/machine-id")
+        boot_entries = glob.glob("/boot/loader/entries/*.conf")
+        for entry in boot_entries:
+            # The boot loader entries in /boot/loader/entries/<machine-id>-<kernel-version>.conf
+            if machine_id.strip() not in os.path.basename(entry):
+                loggerinst.debug("Removing boot entry %s" % entry)
+                os.remove(entry)
```

Comment:
```suggestion
        machine_id = utils.get_file_content("/etc/machine-id").strip()
        boot_entries = glob.glob("/boot/loader/entries/*.conf")
        for entry in boot_entries:
            # The boot loader entries in /boot/loader/entries/<machine-id>-<kernel-version>.conf
            if machine_id not in os.path.basename(entry):
                loggerinst.debug("Removing boot entry %s" % entry)
                os.remove(entry)
```

we don't have to perform strip for every entry